servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultSingleAddressHttpClientBuilder.java

```diff
@@ -219,6 +228,33 @@ private DefaultSingleAddressHttpClientBuilder(@Nullable final U address,
                     proxyAddress != null ? proxyAddress : builder.address);
         }
 
+        Publisher<? extends ServiceDiscovererEvent<R>> discover(
+                PublisherSource.Processor<Throwable, Throwable> processor) {
+            assert builder.address != null : "Attempted to buildStreaming with an unknown address";
+            return builder.serviceDiscoverer.discover(
+                    proxyAddress != null ? proxyAddress : builder.address)
+                    .retryWhen(retryWithConstantBackoffAndJitter(MAX_VALUE, t -> {
```

Comment:
How about starting without providing configurations for this and deferring this API until we have a request? I didn't find a good place for this config, not sure if a client builder or auto-retry strategy provider builder are good for this.

It looks ok to me to always retry `java.net.UnknownHostException`, because this is a well-known JDK exception and not-retrying it means that the discovery publisher will terminate. Therefore, the entire client becomes useless. This is a bit separate discussion, but maybe we need to close the client if SD terminates to clear resources.

For now, users can wrap SD to apply a different retry strategy or map `java.net.UnknownHostException` into something else if their intent is to terminate the discovery publisher.

Reply:
> How about starting without providing configurations for this and deferring this API until we have a request from users? 

If we are using some default strategy that depends on an externally provided implementation (service discoverer in this case), we should always assume there is a case which needs customization. As an example; what if the retry duration has to be changed?

> I didn't find a good place for this config, not sure if a client builder or auto-retry strategy provider builder are good for this.

ServiceDiscoverer retry is not really tied to auto-retry so both of them having some configuration around SD errors. How about adding the method `retryServiceDiscoveryErrors(BiIntFunction<Throwable, ? extends Completable>)`  on the client builder?